### PR TITLE
Fix bug when link to a page doesn't work if it has ? in its title

### PR DESCRIPTION
**Pull request by ViChyavIn**

### CODE REVIEW

**Comment by dometto** [comment]
Thanks for the pr!

File: lib/gollum/views/helpers.rb [hunk 1]
```diff
@@ -61,7 +61,9 @@ def clean_url(*url)
         url.compact!
         return nil if url.empty?
 
-        ::File.join(*url).gsub(%r{/{2,}}, '/')
+        _url = ::File.join(*url)
+        _url = _url.gsub(%r{/{2,}}, '/')
```
Comment (dometto) [inline lib/gollum/views/helpers.rb:65]: The following would be faster:

```ruby
_url.gsub!(%r{/{2,}}, '/')
_url.gsub!(%r{\?}, '%3F')
_url
```

**Review by dometto — CHANGES_REQUESTED** [review]
Small comment related to code performance, otherwise this looks good!

File: lib/gollum/views/helpers.rb (hunk 1 above)
Comment (ViChyavIn) [reply lib/gollum/views/helpers.rb]: Done! https://github.com/ViChyavIn/gollum/commit/7643cd3b421bc0b4b19a27a5a583c76ae32c1b64

**Comment by dometto** [comment]
Thanks!

**Comment by bartkamphorst** [comment]
Do we have tests for these cases?

**Comment by dometto** [comment]
@bartkamphorst no, forgot about that. @ViChyavIn any chance you could submit a regression test guarding against this bug? Sorry for the lateness of this request!

**Comment by ViChyavIn** [comment]
@dometto Added test here: https://github.com/ViChyavIn/gollum/commit/87821d56fa2a9700dbf9bf90bfb24c4b44c2cadb . Should I open another pull request? And please review my test: it seems to work but it's my first time writing tests so I'm not sure if it's written properly.

**Comment by dometto** [comment]
Thanks @ViChyavIn, yes that looks good! A PR would be very welcome.

**Comment by ViChyavIn** [comment]
Done!

**Pull request merged**